public/app/features/explore/Table/TableContainer.tsx

```diff
@@ -88,7 +88,13 @@ export class TableContainer extends PureComponent<Props> {
           frames.map((data, i) => (
             <PanelChrome
               key={data.refId || `table-${i}`}
-              title={dataFrames && dataFrames.length > 1 ? `Table - ${data.name || data.refId || i}` : 'Table'}
+              title={
+                dataFrames && dataFrames.length > 1
+                  ? `Table - ${data.name || data.refId || i}`
+                  : data.name
+                  ? `Table - ${data.name}`
+                  : 'Table'
+              }
```

Comment:
this is getting quite hard to follow, can we move it somewhere?

Reply:
Yeah, with a bit more code it will be more readable:

```typescript

let tableTitle;
if (dataFrames && dataFrames.length > 1) {
  tableTitle = `Table - ${data.name || data.refId || i}`;
} else {
  tableTitle = data.name ? `Table - ${data.name}` : "Table";
}
```